### PR TITLE
27 feature: automatic form submitter

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
- submit the form automatically after x digits are entered into the OTP input field
- update: add the explanation for each of the existing config options to the readme

### CODE REVIEW

File: privacyIDEA-impl/src/main/java/org/privacyidea/context/PIContext.java [hunk 1]
```diff
@@ -17,12 +17,15 @@ public class PIContext extends BaseContext
     private String transactionID = null;
     @Nonnull
     private final String otpFieldHint;
+    @Nullable
+    private final Integer otpLength;
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/java/org/privacyidea/context/PIContext.java:21]: i guess that should be `int` not `Integer`

File: privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm [hunk 2]
```diff
@@ -71,9 +71,11 @@
                         </div>
 
                         <div class="form-element-wrapper">
-                            <input id="pi_otp_input" class="form-element form-field" name="pi_otp_input" type="password" autofocus placeholder=$piContext.getOtpFieldHint() value="">
+                            <input id="pi_otp_input" class="form-element form-field" name="pi_otp_input" type="password"
+                                   onKeyUp='if($piContext.getOtpLength()!=null&&this.value.length===$piContext.getOtpLength())document.getElementById("pi-form-submit-button").click()'
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm:75]: 1. please check if it is possible to extract the `onKeyUp` stuff to a function for readability
2. i think it is better to submit the form via `document.getElementById("privacyidea-form").submit()`

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/InitializePIContext.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package org.privacyidea.action;
 
+import java.nio.channels.ScatteringByteChannel;
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/action/InitializePIContext.java]: you're right. fixed: [3085ca5](https://github.com/privacyidea/shibboleth-plugin/pull/29/commits/3085ca5186a77c43f4835b79f76aec28d538e11d)

File: privacyIDEA-impl/src/main/java/org/privacyidea/context/PIContext.java (hunk 1 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/context/PIContext.java:21]: Integer is right. is returned from external function which changes String to Integer

File: privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm (hunk 2 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm:75]: 1. I can try.
2. This submit button contains an additional event which is returned to the spring flow.

Comment (nilsbehlen) [reply privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm:75]: 👍

Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm]: [22ab4a9](https://github.com/privacyidea/shibboleth-plugin/pull/29/commits/22ab4a9afca649ee08b10c191a61a40672301536)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**